Add RabbitMQ

pabrahamsson opened the pull request:
#### What is this PR About?
A demo of rabbitmq installed as a 3 node cluster.

#### How do we test this?
Follow instructions in README (`oc process -f templates/rabbitmq.yaml | oc apply -f -`)

cc: @redhat-cop/containerize-it

pabrahamsson commented:
Am I supposed to be able to use the `Details` link to see what the issue is? It gives me a `404`, or is that just saying I don't have permissions to view?

sabre1041 commented on rabbitmq/templates/rabbitmq.yaml:
```diff
@@ -0,0 +1,205 @@
+            value: rabbit@$(MY_POD_IP)
+          - name: RABBITMQ_USE_LONGNAME
+            value: "true"
+          image: "172.30.1.1:5000/${APPLICATION_NAME}/${APPLICATION_NAME}:${RABBITMQ_VERSION}"
```
Start in OpenShift 3.6, native Kubernetes resources can make use of ImageStreams

https://docs.openshift.com/container-platform/3.6/release_notes/ocp_3_6_release_notes.html#ocp-36-container-orchestration

It can be enabled by adding the following to the ImageStream and then referencing the name of the imagestream here.
```
  spec:
    lookupPolicy:
      local: true
```

sabre1041 commented on rabbitmq/README.md:
````diff
@@ -0,0 +1,55 @@
+rabbitmq-2         1/1       Running     0          1m
+```
+
+## Verify your RabbitMQ Cluster
````
It does not look like my instances formed a cluster. See below snippets:

```
Cluster status of node 'rabbit@172.17.0.2'
[{nodes,[{disc,['rabbit@172.17.0.2']}]},
 {running_nodes,['rabbit@172.17.0.2']},
 {cluster_name,<<"rabbit@rabbitmq-0.rabbitmq.rabbitmq.svc.cluster.local">>},
 {partitions,[]},
 {alarms,[{'rabbit@172.17.0.2',[]}]}]
```

```
Cluster status of node 'rabbit@172.17.0.3'
[{nodes,[{disc,['rabbit@172.17.0.3']}]},
 {running_nodes,['rabbit@172.17.0.3']},
 {cluster_name,<<"rabbit@rabbitmq-1.rabbitmq.rabbitmq.svc.cluster.local">>},
 {partitions,[]},
 {alarms,[{'rabbit@172.17.0.3',[]}]}]
```

```
Cluster status of node 'rabbit@172.17.0.4'
[{nodes,[{disc,['rabbit@172.17.0.4']}]},
 {running_nodes,['rabbit@172.17.0.4']},
 {cluster_name,<<"rabbit@rabbitmq-2.rabbitmq.rabbitmq.svc.cluster.local">>},
 {partitions,[]},
 {alarms,[{'rabbit@172.17.0.4',[]}]}]
```

pabrahamsson commented:
Thank you for the feedback @sabre1041 
I've updated the ImageStream (very neat feature) and I suspect you were missing the `view` role on the `default` ServiceAccount hence the cluster formation failure. I've updated the template so it creates the necessary `RoleBinding` so no more need to create it upfront.

sabre1041 commented:
LGTM. Thanks!
